File: java/src/main/java/org/ikigaidigital/entity/WithdrawalEntity.java
package org.ikigaidigital.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;

@Entity
@Table(name = "WITHDRAWALS")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class WithdrawalEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;
    @Column(name = "AMOUNT")
    private Double amount;
    @Column(name = "DATE")
    private OffsetDateTime date;
    @ManyToOne
    @JoinColumn(name = "TIMEDEPOSITID")
    private TimeDepositEntity timeDeposit;
}

File: kotlin/src/main/kotlin/org/ikigaidigital/persistence/entity/WithdrawalEntity.kt
package org.ikigaidigital.persistence.entity

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDate

@Entity
@Table(name = "withdrawals")
class WithdrawalEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "time_deposit_id", nullable = false)
    val timeDeposit: TimeDepositEntity,

    @Column(nullable = false, precision = 19, scale = 4)
    val amount: BigDecimal,

    @Column(nullable = false)
    val date: LocalDate
)